core/src/main/java/com/linecorp/armeria/client/limit/ConcurrencyLimit.java

```diff
@@ -54,6 +57,63 @@ static ConcurrencyLimitBuilder builder(int maxConcurrency) {
         return new ConcurrencyLimitBuilder(maxConcurrency == MAX_VALUE ? 0 : maxConcurrency);
     }
 
+    /**
+     * Returns a newly-created {@link ConcurrencyLimit} with the specified {@link IntSupplier}.
+     * {@link IntSupplier#getAsInt()} might be frequently called, so please consider using
+     * {@link SettableIntSupplier} if supplying the value needs a heavy computation. For example:
+     * <pre> {@code
+     * ConcurrencyLimit limit = ConcurrencyLimit.of(new DynamicLimit());
+     *
+     * class DynamicLimit implements IntSupplier {
+     *     private final SettableIntSupplier settableIntSupplier = SettableIntSupplier.of(16);
+     *
+     *     public DynamicLimit() {
```

Comment:
nit:
```suggestion
     *     DynamicLimit() {
```
Because we define the class as package-private, we don't need the `public` modifier. 😄